herd/AArch64Sem.ml

```diff
@@ -71,25 +71,24 @@ module Make
       and is_not_zero v = M.op Op.Ne v V.zero
 
 (* Ordinary access action *)
-      let access_anexp anexp d loc v ac = Act.Access (d,loc,v,AArch64.N,anexp,quad,ac)
-      let access_ord d loc v ac = access_anexp AArch64.Exp d loc v ac
+      let access_anexp anexp d loc v ac attr =
+        Act.Access (d,loc,v,AArch64.N,anexp,quad,ac,attr)
+      let access_ord d loc v ac attr = access_anexp AArch64.Exp d loc v ac attr
 
 (* Basic read, from register *)
-      let mk_read sz an anexp loc v =
+      let mk_read sz an anexp loc v attr =
```

Comment:
I do not understand why memory attributes are part of action. After all memory attributes are present in the page table entry for an access and thus their value can be retrieved from the associated PTE read.

Reply:
Correct, I am not sure how that would work but I'll have a look right away.

Reply:
Having looked a `AArch64Sem.ml` code, the current code makes sense. The difficulty resides in adding yet another argument to the `Access` constructor, we can live with it.

Reply:
We need to the value of the attributes in `herd/machModelChecker.ml` (https://github.com/herd/herdtools7/pull/129/commits/d96e760de272583da8781b8ad3b938fa17aa4528#). Is there a way an easy way to get the `Rpte` event prior to all `M` events to annotate them with attributes? If not do you think it would be worth implementing?

Reply:
Yes, I thought the same, adding one more argument to Access is not making the long constructor any better. I could also turn it into a record to add some structure, but that change would be quite intrusive.

Reply:
I have made specific comment to address the question https://github.com/herd/herdtools7/pull/129#issuecomment-760123055.